examples/signaling-server-libdatachannel-ws/signaling-server.cpp

```diff
@@ -0,0 +1,162 @@
+/**
+ * signaling server example for libdatachannel
+ * Copyright (c) 2024 Tim Schneider
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#include "rtc/rtc.hpp"
+
+#include <csignal>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <thread>
+#include <unordered_map>
+
+#include <nlohmann/json.hpp>
+using nlohmann::json;
+
+
+using namespace std::chrono_literals;
+using std::shared_ptr;
+using std::weak_ptr;
+template <class T> weak_ptr<T> make_weak_ptr(shared_ptr<T> ptr) { return ptr; }
+
+std::string get_user(weak_ptr<rtc::WebSocket> wws) {
+	const auto &ws = wws.lock();
+	const auto path_ = ws->path().value();
+	const auto user = path_.substr(path_.rfind('/') + 1);
+	return user;
+}
+
+void signalHandler(int signum) {
+	std::cout << "Interrupt signal (" << signum << ") received.\n";
+	// terminate program
+	exit(signum);
+}
+
+int main(int argc, char *argv[]) {
+	rtc::WebSocketServerConfiguration config;
+	config.port = 8000;
+	config.enableTls = false;
+	config.certificatePemFile = std::nullopt;
+	config.keyPemFile = std::nullopt;
+	config.keyPemPass = std::nullopt;
+	config.bindAddress = std::nullopt;
+	config.connectionTimeout = std::nullopt;
+
+	// Check command line arguments.
+	for (int i = 1; i < argc; i++) {
+		if (strcmp(argv[i], "--help") == 0) {
```

Comment:
You should not mix C functions in C++ when it is not necessary. Use `std::string`.